packages/zoe/src/typeGuards.js

```diff
@@ -1,18 +1,61 @@
-import { AmountShape } from '@agoric/ertp';
+import {
+  AmountShape,
+  AssetKindShape,
+  DisplayInfoShape,
+  IssuerShape,
+  IssuerKitShape,
+  BrandShape,
+  PaymentShape,
+} from '@agoric/ertp';
 import { M } from '@agoric/store';
 import { TimestampValueShape } from '@agoric/swingset-vat/src/vats/timer/typeGuards.js';
+import { SubscriberShape } from '@agoric/notifier';
+
+// keywords have an initial cap
+export const KeywordShape = M.string();
 
 export const InvitationHandleShape = M.remotable('InvitationHandle');
 export const InvitationShape = M.remotable('Invitation');
 export const InstanceHandleShape = M.remotable('InstanceHandle');
 export const InstallationShape = M.remotable('Installation');
 export const SeatShape = M.remotable('Seat');
 
-export const AmountKeywordRecordShape = M.recordOf(M.string(), AmountShape);
+export const AmountKeywordRecordShape = M.recordOf(KeywordShape, AmountShape);
 export const AmountPatternKeywordRecordShape = M.recordOf(
-  M.string(),
+  KeywordShape,
   M.pattern(),
 );
+export const PaymentKeywordRecordShape = M.recordOf(KeywordShape, PaymentShape);
+export const PaymentPKeywordRecordShape = M.recordOf(
+  KeywordShape,
+  M.eref(PaymentShape),
+);
+export const IssuerKeywordRecordShape = M.recordOf(KeywordShape, IssuerShape);
+export const IssuerPKeywordRecordShape = M.recordOf(
+  KeywordShape,
+  M.eref(IssuerShape),
+);
+export const BrandKeywordRecordShape = M.recordOf(KeywordShape, BrandShape);
+
+export const IssuerRecordShape = M.split(
+  {
+    brand: BrandShape,
+    issuer: IssuerShape,
+    assetKind: AssetKindShape,
+  },
+  { displayInfo: DisplayInfoShape },
+);
+
+export const TermsShape = harden({
+  issuers: IssuerKeywordRecordShape,
+  brands: BrandKeywordRecordShape,
+});
+
+export const InstanceRecordShape = harden({
+  installation: InstallationShape,
+  instance: InstanceHandleShape,
+  terms: M.split(TermsShape, M.record()),
```

Comment:
```suggestion
  terms: M.splitRecord(TermsShape, harden({}), M.record()),
```

Reply:
@erights Is there an explanation of `splitRecord` and `splitArray` somewhere? I worked out that it's basically `M.splitRecord(requiredFieldsShape, optionalFieldsShape, restFieldsShape)` and `M.splitArray(harden([firstItemShape, secondItemShape, …]), harden([firstOptionalItemShape, …]), restItemsShape)` (with the third argument of each being a record/array shape rather than an array item/record field value shape?), but [patternMatchers.js](https://github.com/Agoric/agoric-sdk/blob/162fdf2a51c54e2da89c8ed35f06cab09fb013d2/packages/store/src/patterns/patternMatchers.js#L1765-L1774) is very terse and doesn't really explain that.

In particular, I don't know if `M.splitRecord(TermsShape, harden({}), M.record())` should be interpreted as "_a record matching `TermsShape` with arbitrary extra fields_" or as "_a record matching TermsShape with arbitrary **record-valued** extra fields_".

Reply:
Ah, found it: [packages/store/src/types.js](https://github.com/Agoric/agoric-sdk/blob/162fdf2a51c54e2da89c8ed35f06cab09fb013d2/packages/store/src/types.js#L570-L593).

Reply:
Better:

```suggestion
  terms: M.splitRecord(TermsShape),
```